client/js/libs/jquery/stickyscroll.js

```diff
@@ -1,67 +1,36 @@
-/*!
- * stickyscroll
- * https://github.com/erming/stickyscroll
- * v2.2.0
- */
 (function($) {
-	$.fn.sticky = function() {
-		if (this.size() > 1) {
-			return this.each(function() {
-				$(this).sticky(options);
-			});
-		}
+	$.fn.unsticky = function() {
+		return this.unbind(".sticky");
+	};
 
-		var isBottom = false;
+	$.fn.sticky = function() {
 		var self = this;
-
-		this.unbind(".sticky");
-		this.on("beforeAppend.sticky", function() {
-			isBottom = isScrollBottom.call(self);
-		});
-
-		this.on("afterAppend.sticky", function() {
-			if (isBottom) {
-				self.scrollBottom();
-			}
-		});
-
-		var overflow = this.css("overflow-y");
-		if (overflow == "visible") {
-			overflow = "auto";
-		}
-		this.css({
-			"overflow-y": overflow
-		});
-
-		$(window).unbind(".sticky");
-		$(window).on("resize.sticky", function() {
-				self.scrollBottom();
-		});
-
-		this.scrollBottom();
-		return this;
+		var stuckToBottom = true;
+
+		lastHeight = self.outerHeight(true);
+
+		self
+			.on("scroll.sticky", function(e) {
+				stuckToBottom = self.isScrollBottom();
+			})
+			.on("msg.sticky", function() {
+				if (stuckToBottom) {
+					self.scrollBottom();
+				}
+			})
+			.scrollBottom();
+
+		return self;
 	};
 
 	$.fn.scrollBottom = function() {
-		return this.each(function() {
-			$(this).animate({scrollTop: this.scrollHeight}, 0);
-		});
-	};
-
-	$.fn.isScrollBottom = isScrollBottom;
-
-	function isScrollBottom() {
-		if ((this.scrollTop() + this.outerHeight() + 1) >= this.prop("scrollHeight")) {
-			return true;
-		} else {
-			return false;
-		}
+		var el = this[0];
+		this.scrollTop(this[0].scrollHeight);
```

Comment:
Crashes on private instances with `Uncaught TypeError: Cannot read property 'scrollHeight' of undefined`.

Seems like `.sticky()` is called on an empty jQuery set. Probably should loop over it since this is how jQuery recommends to do it, so we don't get surprises later if we try to sticky multiple things at once.

```
a.fn.scrollBottom	@	stickyscroll.js:28
a.fn.sticky	@	stickyscroll.js:21
(anonymous function)	@	lounge.js:670
n.event.dispatch	@	jquery.js:4409
r.handle	@	jquery.js:4095
n.event.trigger	@	jquery.js:4324
(anonymous function)	@	jquery.js:4875
n.extend.each	@	jquery.js:375
n.fn.n.each	@	jquery.js:139
n.fn.extend.trigger	@	jquery.js:4874
n.fn.(anonymous function)	@	jquery.js:7461
(anonymous function)	@	lounge.js:105
Emitter.emit	@	socket.io.js:5657
Socket.onevent	@	socket.io.js:5183
Socket.onpacket	@	socket.io.js:5141
(anonymous function)	@	socket.io.js:5522
Emitter.emit	@	socket.io.js:5657
Manager.ondecoded	@	socket.io.js:4668
(anonymous function)	@	socket.io.js:5522
Emitter.emit	@	socket.io.js:2556
Decoder.add	@	socket.io.js:6156
Manager.ondata	@	socket.io.js:4658
(anonymous function)	@	socket.io.js:5522
Emitter.emit	@	socket.io.js:2556
Socket.onPacket	@	socket.io.js:459
(anonymous function)	@	socket.io.js:276
Emitter.emit	@	socket.io.js:2556
Transport.onPacket	@	socket.io.js:892
callback	@	socket.io.js:1766
(anonymous function)	@	socket.io.js:3558
exports.decodePayloadAsBinary	@	socket.io.js:3557
exports.decodePayload	@	socket.io.js:3325
Polling.onData	@	socket.io.js:1770
(anonymous function)	@	socket.io.js:1330
Emitter.emit	@	socket.io.js:2556
Request.onData	@	socket.io.js:1491
Request.onLoad	@	socket.io.js:1572
xhr.onreadystatechange	@	socket.io.js:1444
```